Source/Extensions/Blazorise.Cropper/CropperViewer.razor.cs

```diff
@@ -0,0 +1,75 @@
+﻿#region Using directives
+using System;
+using System.Threading.Tasks;
+using Blazorise.Infrastructure;
+using Microsoft.AspNetCore.Components;
+#endregion
+
+namespace Blazorise.Cropper
+{
+    /// <summary>
+    /// The CropperViewer interface provides properties and methods for manipulating the layout and presentation of cropper elements.
+    /// </summary>
+    public partial class CropperViewer : BaseComponent, IDisposable
+    {
+        #region Members
+
+        private readonly EventCallbackSubscriber<Cropper> cropperInitialized;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// A default <see cref="CropperViewer"/> constructor.
+        /// </summary>
+        public CropperViewer()
+        {
+            cropperInitialized = new EventCallbackSubscriber<Cropper>( EventCallback.Factory.Create<Cropper>( this, OnCropperInitialized ) );
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <inheritdoc/>
+        protected override Task OnParametersSetAsync()
+        {
+            cropperInitialized.SubscribeOrMove( CropperState?.CropperInitialized );
+
+            return base.OnParametersSetAsync();
+        }
+
+        /// <inheritdoc/>
+        protected override void Dispose( bool disposing )
+        {
+            if ( disposing )
+            {
+                cropperInitialized.Dispose();
+            }
+
+            base.Dispose( disposing );
+        }
+
+        private async Task OnCropperInitialized( Cropper cropper )
+        {
+            if ( cropper?.JSModule != null )
+            {
+                await cropper.JSModule.InitializeViewer( cropper.ElementRef, cropper.ElementId, ElementRef, ElementId, new
+                {
+                } );
+            }
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Provides a shared state and syncronization context between the cropper and cropper viewer.
+        /// </summary>
+        [Parameter] public CropperState CropperState { get; set; }
```

Comment:
Doesn't it make more sense to 'bind' a viewer to a cropper instead of passing the state? Now the user needs to have knowledge of the inner workings of the Cropper and Viewer.

```
<Cropper @ref="cropper" />
<CropperViewer Target="@cropper" />
```

Reply:
I've tried that. The problem is that the order of rendering is important. When `CropperViewer` is rendered before `Cropper` then the `Target` will be null and the viewer will not be created properly.

Reply:
ok that is less indeal indeed